### PR TITLE
feat(codecs): implement xsd:integer codec

### PR DESCRIPTION
**Pull request by mark-dropbear**
- Add `XsdInteger` extension type as a wrapper for `BigInt`.
- Implement `XsdIntegerCodec` with support for XSD 1.1 compliant lexical forms.
- Register `integer` in the `xsd` facade and export from `lib/xsd.dart`.
- Add comprehensive unit tests for parsing, encoding, and whitespace handling.
- Update `README.md` status for `xsd:integer`.

### CODE REVIEW

File: lib/src/codecs/integer.dart [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    if (!regex.hasMatch(input)) {
+      throw FormatException('Invalid xsd:integer lexical form: $input');
+    }
+
+    try {
+      return XsdInteger(BigInt.parse(input));
+    } on FormatException {
+      throw FormatException('Invalid xsd:integer lexical form: $input');
+    }
+  }
```
[the remark this answers is not in the document]
Comment (mark-dropbear) [reply lib/src/codecs/integer.dart:34]: @gemini-code-assist

You are incorrect here. Consider the following:

```dart
void main() { 
  print('BigInt.parse("0x10"): ${BigInt.tryParse("0x10")}');                                                                                   │
  print('BigInt.parse("10"): ${BigInt.tryParse("10")}'); 
} 
```

I have added an additional test now to capture this scenario.
